OpenSim/Common/Component.cpp

```diff
@@ -1526,11 +1529,39 @@ void Component::dumpConnections() const {
 
 
 void Component::initComponentTreeTraversal(const Component &root) const {
-    // Going down the tree, node is followed by all its
-    // children in order, last child's successor is the parent's successor.
+    // Going down the tree, this node is followed by all its children.
+    // The last child's successor (next) is the parent's successor.
+
+    const size_t nmsc = _memberSubcomponents.size();
+    const size_t npsc = _propertySubcomponents.size();
+    const size_t nasc = _adoptedSubcomponents.size();
+
+    // If this isn't the root component and it has no parent, then
+    // this is an orphan component and we likely failed to call 
+    // finalizeFromProperties on the root OR this is a clone that
+    // has not been added to the root (in which case would have a parent).
+    if ((this != &root) && !this->hasParent() ) {
+        OPENSIM_THROW(ComponentIsAnOrphan, getName(), getConcreteClassName());
+    }
+
+    if ((this == &root) && !(nmsc + npsc + nasc)) {
+        ComponentIsRootWithNoSubcomponents ex(__FILE__, __LINE__, __func__,
+            getName(), getConcreteClassName());
+        if (getConcreteClassName() == "Model") {
```

Comment:
Might we want to relax this check to catch any ModelComponent? I'm thinking about a situation like the Device in the hopper example: it's not a Model (it extends ModelComponent), but it would be root and a user might want to traverse the tree without attaching it to a model first.